builtin/mainmenu/dlg_register.lua

```diff
@@ -0,0 +1,128 @@
+--Minetest
+--Copyright (C) 2022 rubenwardy
+--
+--This program is free software; you can redistribute it and/or modify
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+--------------------------------------------------------------------------------
+
+local function register_formspec(dialogdata)
+	local title = fgettext("Joining $1", dialogdata.address)
+	local buttons_y = 4 + 1.3
+	if dialogdata.error then
+		buttons_y = buttons_y + 0.8
+	end
+
+	local retval = {
+		"formspec_version[4]",
+		"size[8,", tostring(buttons_y + 1.175), "]",
+		"set_focus[", (dialogdata.name ~= "" and "password" or "name"), "]",
+		"label[0.375,0.8;", core.formspec_escape(title), "]",
+		"field[0.375,1.575;7.25,0.8;name;", core.formspec_escape(fgettext("Name")), ";",
+				core.formspec_escape(dialogdata.name), "]",
+		"pwdfield[0.375,2.875;7.25,0.8;password;", core.formspec_escape(fgettext("Password")), "]",
+		"pwdfield[0.375,4.175;7.25,0.8;password_2;", core.formspec_escape(fgettext("Confirm Password")), "]"
+	}
+
+	if dialogdata.error then
+		table.insert_all(retval, {
+			"style_type[label;textcolor=", mt_color_orange, "]",
+			"label[0.375,", tostring(buttons_y - 0.6), ";", core.formspec_escape(dialogdata.error), "]",
```

Comment:
I think a `box[]` would be nice around the error message.

Reply:
Which color?  A darker red with white text?